store/postgres/migrations/2019-03-07-171355_update_history_storage/up.sql

```diff
@@ -0,0 +1,178 @@
+-- Change log_entity_event such that we do not record history entries for
+-- the 'subgraphs' subgraph and for operations that don't actually change
+-- the data of an entity
+CREATE OR REPLACE FUNCTION log_entity_event()
+    RETURNS trigger AS
+$$
+DECLARE
+    event_id INTEGER;
+    new_event_id INTEGER;
+    is_reversion BOOLEAN := FALSE;
+    operation_type INTEGER := 10;
+    event_source  VARCHAR;
+    subgraph VARCHAR;
+    entity VARCHAR;
+    entity_id VARCHAR;
+    data_before JSONB;
+    data_after JSONB;
+BEGIN
+    -- Get operation type and source
+    IF (TG_OP = 'INSERT') THEN
+        operation_type := 0;
+        event_source := NEW.event_source;
+        subgraph := NEW.subgraph;
+        entity := NEW.entity;
+        entity_id := NEW.id;
+        data_before := NULL;
+        data_after := NEW.data;
+    ELSIF (TG_OP = 'UPDATE') THEN
+        operation_type := 1;
+        event_source := NEW.event_source;
+        subgraph := OLD.subgraph;
+        entity := OLD.entity;
+        entity_id := OLD.id;
+        data_before := OLD.data;
+        data_after := NEW.data;
+    ELSIF (TG_OP = 'DELETE') THEN
+        operation_type := 2;
+        event_source := current_setting('vars.current_event_source', TRUE);
+        subgraph := OLD.subgraph;
+        entity := OLD.entity;
+        entity_id := OLD.id;
+        data_before := OLD.data;
+        data_after := NULL;
+    ELSE
+        RAISE EXCEPTION 'unexpected entity row operation type, %', TG_OP;
+    END IF;
+
+    -- We do not need any history for the 'subgraphs' subgraph nor
+    -- do we need to record changes that do not change anything
+    IF subgraph = 'subgraphs' OR data_before = data_after THEN
+        RETURN NULL;
+    END IF;
+
+    IF event_source = 'REVERSION' THEN
+        is_reversion := TRUE;
+    END IF;
+
+    SELECT id INTO event_id
+    FROM event_meta_data
+    WHERE db_transaction_id = txid_current();
+
+    new_event_id := null;
+
+    IF event_id IS NULL THEN
```

Comment:
I know this was here before, but now that I look at it, I don't understand what this (and the `SELECT` query just above) does. Can we add a comment?

Reply:
Yeah, a comment is in order. It's making sure that we have a `meta_data_event` for the current transaction, but that we only create one per transaction (so that all `entity_history` entries that get created in the txn are tied to the same event)